Ports/.port_include.sh

```diff
@@ -394,16 +437,30 @@ func_defined pre_patch || pre_patch() {
 }
 
 func_defined patch_internal || patch_internal() {
+    if [ -n "${IN_SERENITY_PORT_DEV:-}" ]; then
+        return
+    fi
+
     # patch if it was not yet patched (applying patches multiple times doesn't work!)
-    if [ -z "${IN_SERENITY_PORT_DEV:-}" ] && [ -d "${PORT_META_DIR}/patches" ]; then
+    if [ -d "${PORT_META_DIR}/patches" ]; then
         for filepath in "${PORT_META_DIR}"/patches/*.patch; do
             filename=$(basename $filepath)
-            if [ ! -f "$workdir"/.${filename}_applied ]; then
+            if [ -f "$workdir"/.${filename}_applied ]; then
+                continue
+            fi
+
+            if [ -e "${workdir}/.git" ]; then
+                run git am --keep-cr --keep-non-patch "${filepath}"
+            else
                 run patch -p"$patchlevel" < "$filepath"
                 run touch .${filename}_applied
             fi
         done
     fi
+
+    if [ -e "${workdir}/.git" ]; then
+        run git tag -f original
```

Comment:
Why is `original` _after_ patching? Maybe we should use `original` (or `upstream`, etc.) and `patched` as tags to differentiate between the revisions?

Reply:
That naming is unchanged from `./package.sh dev`, where we had/have "import[ed tarball]", "original [state]" (i.e. before the user started `dev`) and the current state (which is wherever `HEAD` happens to be at).

I'll try and think of some better names.

Reply:
Ah then I mainly disagree with the original choices, let's see if we can come up with something in this PR. Maybe `source` (as downloaded from `$files`) and `patched` (state after patches)?